test/e2e/data/infrastructure-gcp/clusterclass-quick-start.yaml

```diff
@@ -0,0 +1,167 @@
+      schema:
+        openAPIV3Schema:
+          type: string
+          default: n1-standard-2
+  patches:
+    - name: region
+      definitions:
+        - selector:
+            apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+            kind: GCPClusterTemplate
+            matchResources:
+              infrastructureCluster: true
+          jsonPatches:
+            - op: add
+              path: /spec/template/spec/region
+              valueFrom:
+                variable: region
+    - name: controlPlaneMachineType
+      definitions:
+        - selector:
+            apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+            kind: GCPMachineTemplate
+            matchResources:
+              controlPlane: true
+          jsonPatches:
+            - op: replace
+              path: /spec/template/spec/instanceType
+              valueFrom:
+                variable: controlPlaneMachineType
+    - name: workerMachineType
+      definitions:
+        - selector:
+            apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+            kind: GCPMachineTemplate
+            matchResources:
+              machineDeploymentClass:
+                names:
+                  - default-worker
+          jsonPatches:
+            - op: replace
+              path: /spec/template/spec/instanceType
+              valueFrom:
+                variable: workerMachineType
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: GCPClusterTemplate
+metadata:
+  name: quick-start
+spec:
+  template:
+    spec:
+      project: "${GCP_PROJECT}"
+      region: "${GCP_REGION}"
+      network:
+        name: "${GCP_NETWORK_NAME}"
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlaneTemplate
+metadata:
+  name: quick-start-control-plane
+spec:
+  template:
+    spec:
+      kubeadmConfigSpec:
+        useExperimentalRetryJoin: true
+        initConfiguration:
+          nodeRegistration:
+            name: '{{ ds.meta_data.local_hostname.split(".")[0] }}'
+            kubeletExtraArgs:
+              cloud-provider: gce
+        clusterConfiguration:
+          apiServer:
+            timeoutForControlPlane: 20m
+            extraArgs:
+              cloud-provider: gce
+          controllerManager:
+            extraArgs:
+              cloud-provider: gce
+              allocate-node-cidrs: "false"
+        joinConfiguration:
+          nodeRegistration:
+            name: '{{ ds.meta_data.local_hostname.split(".")[0] }}'
+            kubeletExtraArgs:
+              cloud-provider: gce
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: GCPMachineTemplate
+metadata:
+  name: quick-start-control-plane
+spec:
+  template:
+    spec:
+      instanceType: REPLACEME
+      image: "${IMAGE_ID}"
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: GCPMachineTemplate
+metadata:
+  name: quick-start-worker-machinetemplate
+spec:
+  template:
+    spec:
+      instanceType: REPLACEME
```

Comment:
this is correct?
we can use `GCP_CONTROL_PLANE_MACHINE_TYPE` or `GCP_NODE_MACHINE_TYPE`

Reply:
A great question. Yes, we can definitely use the variables.

I am testing ClusterClass' variables and patching mechanism here. The value, `REPLACEME`, doesn't matter as it will be replaced with [variables defined in Cluster spec](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/568/files#diff-2700dbb61b8d021524d25ba8cac949488a62df05587f873a8c930753864cbd3aR26-R29) using [the patching](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/568/files#diff-0f39d30babba1368317e93cb83d174c1e5d025a364e0383b1eed577aad9b2a18R68-R93).

Reply:
cool, thanks for the clarification